packages/experience/src/pages/IdentifierRegister/index.tsx

```diff
@@ -1,21 +1,23 @@
-import { AgreeToTermsPolicy, experience } from '@logto/schemas';
+import { AgreeToTermsPolicy, experience, SignInMode } from '@logto/schemas';
 import { useTranslation } from 'react-i18next';
 import { Navigate } from 'react-router-dom';
 
 import FocusedAuthPageLayout from '@/Layout/FocusedAuthPageLayout';
 import IdentifierRegisterForm from '@/components/IdentifierRegisterForm';
+import { useSieMethods } from '@/hooks/use-sie';
 import { identifierInputDescriptionMap } from '@/utils/form';
 
 import useIdentifierSignUpMethods from './use-identifier-sign-up-methods';
 
 const IdentifierRegister = () => {
   const { t } = useTranslation();
   const signUpMethods = useIdentifierSignUpMethods();
+  const { signInMode } = useSieMethods();
 
   /**
    * Fallback to sign-in page if no sign up methods are available (not allowed to create an account).
    */
-  if (signUpMethods.length === 0) {
+  if (signInMode === SignInMode.SignIn || signUpMethods.length === 0) {
```

Comment:
Better add some comment here to explain `signInMode === SignInMode.SignIn` indicates the user registration is not enabled

Reply:
Added.